src/vivarium/framework/lookup/table.py

```diff
@@ -136,16 +139,23 @@ def __init__(
 
         self.parameter_columns_with_edges = param_cols_with_edges
 
-        extra_columns = self.data.columns.difference(
+        if not isinstance(self.data, pd.DataFrame):
+            raise ValueError(
+                f"The data used to create an interpolated LookupTable must be a DataFrame, but {type(self.data)} was provided instead."
+            )
+
+        required_cols = (
             set(self.key_columns)
             | set([col for p in self.parameter_columns_with_edges for col in p])
             | set(self.value_columns)
         )
+        extra_columns = self.data.columns.difference(list(required_cols))
 
         if not self.value_columns:
             self.value_columns = list(extra_columns)
         else:
-            self.data = self.data.drop(columns=extra_columns)
+            if isinstance(self.data, pd.DataFrame):
```

Comment:
This could be `elif`, but if you are throwing if `self.data` is not a DataFrame above, you don't need this check here.